### PR TITLE
Fix issues with alert (windows)

### LINKED ISSUE #23483: [Bug]: Windows `10min_cpu_usage` alert attaches to `service.*_cpu_utilization` charts and fires false CRITICAL
## Bug report summary

On Windows, the stock `10min_cpu_usage` health template (intended for the whole-machine `system.cpu` chart) instantiates on per-service apps.plugin charts, producing false "System CPU utilization" CRITICAL alerts.

Observed: the alert fired on chart `service.not-services_cpu_utilization` at ~226–259 %, using the template's `system.cpu` lookup (`irq,user,system,dpc`) and thresholds (warn 85 % / crit 95 %). That chart's title is "Windows Service CPU utilization (100% = 1 core)", so a value >100 % is correct and expected whenever a service group uses more than one core — the whole-machine thresholds do not apply.

## Evidence (netdata v2.10.4, Windows 11, i5-1245U)

- `/api/v1/alarms` showed exactly one alarm: `service.not-services_cpu_utilization.10min_cpu_usage` (CRITICAL, value 226–259 %), with `lookup_dimensions: irq,user,system,dpc` and `warn`/`crit` of 85/95 % — i.e. the Windows `10min_cpu_usage` template from `health.d/cpu.conf` (`on: system.cpu`).
- No `system.cpu.10min_cpu_usage` alarm existed at all, even though `system.cpu` was reading ~100 % at the time (it would have been CRITICAL had it been instantiated). So the template attached to the service chart *instead of* `system.cpu`.

## Expected behavior

The `10min_cpu_usage` template should attach only to the `system.cpu` context (whole-machine, 0–100 %), not to `service.*_cpu_utilization` charts whose unit is "100 % = 1 core".

## Workaround (for anyone hitting this)

`netdata.conf` → `[health]` → `enabled alarms = !10min_cpu_usage *` disables the misfiring template on the affected node.

## Environment

- Netdata v2.10.4, Windows 11 (build 26300), HP ZBook Firefly G9, 12th Gen Intel Core i5-1245U.
- Stock health config: `usr/lib/netdata/conf.d/health.d/cpu.conf` (Windows `10min_cpu_usage` template, `on: system.cpu`, `host labels: _os=windows`).

### PR DESCRIPTION
**Pull request by thiagoftsm**
##### Summary
Fix https://github.com/netdata/netdata/issues/23483

##### Additional Information

<details> <summary>For users: How does this change affect me?</summary>
  
</details>

---
## Summary by cubic
Prevents same-name alerts from attaching to every chart when a non-head rule lacks the 'on' field. Previously, such a rule could match all charts; now each rule is validated and a missing or empty 'on' is either rejected (dynamic config) or matches nothing (runtime), preserving per-OS rule sets.

- Validate each rule: missing 'on' (context/chart), non-positive update frequency, no lookup/calc/warn/crit, or non-finite delay multiplier.
- For static files, log and skip only the invalid rule; valid same-name rules remain.
- For dynamic config, reject the payload with a detailed reason, and log a succinct rejection during startup replay.
- Make matching fail closed: templates require on.context and instances require on.chart; a missing value never matches.
- Fix delay multiplier parsing to always fall back to the default on invalid or non-positive values, including when a later 'multiplier' token is invalid.
- Add unit tests for empty/missing 'on', fail-closed matching, per-rule validation, multiplier parsing, and dynamic-config error reporting; update docs to note invalid static rules are skipped while valid same-name rules remain.

<sup>Written for commit 568dda956c1f1c92173f3e94b29024aa41d52e97. Summary will update on new commits.</sup>

<a href="https://cubic.dev/pr/netdata/netdata/pull/23559?utm_source=github" target="_blank" rel="noopener noreferrer" data-no-image-dialog="true"><picture><source media="(prefers-color-scheme: dark)" srcset="https://www.cubic.dev/buttons/review-in-cubic-dark.svg"><source media="(prefers-color-scheme: light)" srcset="https://www.cubic.dev/buttons/review-in-cubic-light.svg">Review in cubic</picture></a>

## Summary by CodeRabbit

* **Bug Fixes**
  * Alert chains now validate every rule for required match targets, update frequency, expressions, and finite delay multipliers.
  * Invalid rules are rejected with details identifying the affected rule, while valid same-name rules remain available.
  * Rules missing required chart or context targets no longer match charts unexpectedly.
  * Dynamic Configuration consistently rejects empty match targets and reports rejected prototype changes.
  * Invalid delay multipliers now safely use the default value.

* **Documentation**
  * Clarified first-match behavior for same-name alert rules and per-OS configurations.

### CODE REVIEW

File: src/health/health-config-unittest.c [hunk 1]
```diff
@@ -872,6 +889,288 @@
...
+        fprintf(stderr, "FAILED [dyncfg rejection detail]: code=%d response='%s'\n", code, buffer_tostring(result));
+        return 1;
+    }
+
+    (*passed)++;
+    return 0;
+}
+
+static int test_file_invalid_same_name_rule_is_skipped(int *passed) {
+    char filename[] = "/tmp/netdata-health-config-unittest-XXXXXX";
```
[the remark this answers is not in the document]
Comment (thiagoftsm) [reply src/health/health-config-unittest.c]: This was fixed with latest commit

**Review by stelfrag — APPROVED** [review]
